src/components/dashboard/SendByQR.js

```diff
@@ -55,18 +76,42 @@ const SendByQR = ({ screenProps }: Props) => {
 
   const onDismissDialog = () => setQRDelay(QR_DEFAULT_DELAY)
 
+  const gotoSend = useCallback(
+    async code => {
+      const { route, params } = await routeAndPathForCode('sendByQR', code)
+      log.info({ code })
+      fireEvent(QR_SCAN, { type: 'send' })
+      push(route, params)
+    },
+    [push],
+  )
   const handleScan = useCallback(
     async data => {
       if (data) {
+        let code
         try {
           const decoded = decodeURI(data)
-          const paramsUrl = extractQueryParams(decoded)
-          const code = readCode(paramsUrl.code)
-          const { route, params } = await routeAndPathForCode('sendByQR', code)
 
-          log.info({ code })
-          fireEvent(QR_SCAN, { type: 'send' })
-          push(route, params)
+          const address = get(decoded.match(/0x[a-fA-F0-9]{40}/), '0')
```

Comment:
A more readable way: 
```
const ethAddressRegex = /0x[a-fA-F0-9]{40}/ 
const address = get(decoded.match(ethAddressRegex), '0') 
```

Reply:
@sirpy hmm. maybe add utility (e.g. `getEthAddress`) somewhere in the wallet/utils ?

```javascript
const ethAddressRegex = /0x[a-fA-F0-9]{40}/ 
export const extractEthAddress = value => get(value.match(ethAddressRegex), '0', null) 
```

Reply:
@johnsmith-gooddollar can you do that and merge?

Reply:
@sirpy yes, will do